### ISSUE TITLE
Fix validated PR #3 review findings in Stage 0 workflow

### ISSUE THREAD

**Issue by TychoHenzen** [TychoHenzen/Pensieve#4]
## Outcome

The Stage 0 trainability investigation executes the declared bounded workflow against the exact failed stability report and emits a typed, truthful result. Invalid or stale inputs fail before any recalibration decision.

## Scope

Included:

- Wire the trainability command to the existing Stage 0 dataset, report, probe, arm, classification, and progress contracts.
- Correct checkpoint geometry, deterministic model-loading policy, causal metrics, overfit gates, eligibility, and stability-report compatibility.
- Make no-update, gradient, and EGGROLL arms use their declared update semantics and budgets.
- Correct the Idle renderer contract and stale repository guidance.
- Add focused regression tests for every finding, including a real command-path test that does not replace the investigation runner.

Non-goals:

- No new model architecture, dataset, model revision, optimizer policy, or Stage 1+ behavior.
- No automatic full training authorization or production recalibration.
- No changes outside the 11 validated PR #3 findings listed in this issue.

## Implementation notes

Research evidence:

- train/run_stage0_trainability.py:82 currently returns inconclusive with investigation_incomplete and no report evidence. main calls it at line 210, while train/stage0_trainability.py already provides typed report models, fixed record selection, probes, arm runners, classifications, and progress types.
- train/stage0_trainability.py:462 validates only status, implementation SHA, and held-out count. The canonical train/eggroll_stability.py contract contains the complete asset identity, implementation source set, EGGROLL parameters, initialization seed, development count, and checkpoints.
- train/stage0_trainability.py:668 accepts an overfit attempt after a scalar loss ratio. The typed classification model already exposes exact accuracy, first-token accuracy, and finite-value checks, but the command path does not produce the required complete metrics.
- train/stage0_trainability.py:873 evaluates eight records and passes len(metric_components) into StabilityMetrics, although StabilityMetrics requires exactly 64 problems.
- train/stage0_trainability.py:1146 treats missing causal status as acceptable, allows direction_mismatch, and allows direction_mismatch method status. Recalibration must require a passed causal probe and a viable method.
- train/stage0_trainability.py:1517 always calls LatentCoreTrainer.train_step while ignoring arm_kind and use_eggroll_batches. train/eggroll_trainer.py already exposes train_fitness_batch for the EGGROLL path.
- eval/gate/latent_eval.py:82 replaces the requested slot_count with checkpoint slot-query geometry. tests/test_latent_eval.py:44 currently asserts this override and must become a mismatch regression.
- eval/stage0_identity.py:283 assigns CUBLAS_WORKSPACE_CONFIG locally but does not enforce the shared deterministic runtime policy. The shared train/standalone_checkpoint.py:33 contract also disables nondeterministic algorithms, TF32, and cuDNN benchmarking.
- eval/stream/render.py:197 raises TypeError for Idle, while the stream contract and requested behavior require ValueError.
- AGENTS.md:138 still names eval/gate.py, eval/run/, and eval/generators/ as current paths, and line 128 names nonexistent step_from_checkpoint. Update only stale references identified by this issue.

Implementation direction:

- Reuse the existing typed report and strict stability contracts. Build the real runner around fixed selections of one overfit record, 32 training records, and 64 held-out records, with deterministic fresh state for each arm and flushed progress records.
- Route no-update, gradient, and EGGROLL arms through their declared trainer semantics. Keep the bounded investigation diagnostic only and preserve the distinction between trainability evidence and full-training authorization.
- Reject incompatible checkpoint slot geometry instead of silently changing the caller's requested configuration. Centralize deterministic runtime setup before model construction and record the effective runtime identity.
- Extend regression tests at the existing focused test files. Update tests that currently encode the invalid behavior, rather than weakening the contract.

Classification evidence:

- Prio 2 - Urgent. The hard-coded command is a blocker for the Stage 0 milestone, and invalid eligibility or stale identity validation can authorize unsafe or stale recalibration. This matches the live description, serious issues or important milestones that heavily impact progress.
- Effort 8 - Huge. This is a complex cross-module repair across the trainability command, Stage 0 training/evaluation modules, stream rendering, repository guidance, and several focused test files. Existing typed contracts and callers bound the direction, so the 11 criteria form one independently deliverable gate repair rather than an Effort 13 epic.
- Standard labels: bug matches Something isn't working for the ten behavioral findings. documentation matches Improvements or additions to documentation for the stale AGENTS.md paths.
- No linked sub-issues are needed. The criteria share one report identity and one Stage 0 gate workflow, and splitting them would prevent independent acceptance of the end-to-end investigation contract.

## Acceptance criteria

- [ ] The trainability CLI runs the real bounded investigation from a failed compatible stability report. Its final JSON contains the current implementation and asset identities, selected records, overfit attempts, causal probes, arm results, elapsed time, and the actual overall status. It never emits zero-filled placeholder identities or investigation_incomplete for a normal execution path.
- [ ] The command emits flushed progress records for the declared overfit, causal, and arm checkpoints, preserves exclusive output-file behavior, and maps bounded_trainability_observed to exit code 0 while non-viable or inconclusive outcomes return exit code 1.
- [ ] The stability-report loader rejects any status, asset-selection, implementation-source, EGGROLL-configuration, initialization-seed, development-count, or checkpoint mismatch. It accepts only a complete failed report compatible with the current canonical configuration.
- [ ] The no-update arm performs no optimizer update, the gradient arm uses gradient updates, and the EGGROLL arm uses EGGROLL fitness batches with the declared batch boundaries. Each arm starts from an isolated fresh state and reports the correct consumed-example and optimizer-call counts.
- [ ] Causal evaluation measures exactly 64 selected held-out problems and constructs valid StabilityMetrics. A short or non-finite evaluation is reported as failed or inconclusive evidence, not as a valid 8-problem metric.
- [ ] An overfit probe passes only when the selected attempt has finite metrics, the declared loss reduction, exact-answer success, and first-token success. Missing or non-finite answer metrics cannot produce a passed probe.
- [ ] Recalibration eligibility is false unless overfit status is passed, method status is viable, and causal status is exactly passed. Missing causal status and direction_mismatch are ineligible and are named in the missing-prerequisites evidence.
- [ ] Latent checkpoint loading honors the requested slot_count. A checkpoint whose slot-query count differs from that request is rejected before subject construction instead of silently overriding the request.
- [ ] Frozen Qwen loading uses the shared deterministic runtime contract before model construction: the effective CUBLAS configuration is explicit, deterministic algorithms are enabled, TF32 is disabled, and cuDNN benchmarking is disabled. The loader does not apply a conflicting loader-local runtime assignment.
- [ ] eval.stream.render.render_event raises ValueError for Idle, its message remains actionable, and the affected renderer tests assert the contract.
- [ ] AGENTS.md contains only current repository paths for the affected workflow and no stale step_from_checkpoint reference. Focused regression tests cover every corrected finding, including the real trainability command path without monkeypatching _run_investigation.

## Verification

- pytest -q tests/test_run_stage0_trainability.py tests/test_stage0_trainability_reports.py tests/test_stage0_trainability_identity.py tests/test_latent_eval.py tests/gate/test_stage0_latent_evaluation.py tests/test_stage0_identity.py tests/stream/test_render.py
- pytest -q
- ruff check .
- ruff format --check .
- pyright
- openspec validate --all --strict --no-interactive
- Run the CLI against a temporary failed stability report and inspect the final JSON and JSONL progress output for non-placeholder identities, complete probe and arm evidence, finite values, and the expected exit code.
